src/main/java/global/govstack/mocksris/service/PaymentHubService.java

```diff
@@ -0,0 +1,272 @@
+    httpHeaders.add("Purpose", "USCT Payment");
+    httpHeaders.add("X-CorrelationID", requestID);
+    httpHeaders.add("Platform-TenantId", paymentHubProperties.tenant());
+    httpHeaders.add("X-Program-ID", paymentHubProperties.programId());
+    httpHeaders.add("X-Signature", signOrderPaymentRequest(requestID, body));
+    httpHeaders.add("type", "raw");
+
+    var response =
+        new RestTemplate(requestFactory)
+            .exchange(
+                paymentHubProperties.bulkConnectorURL() + "/batchtransactions?type=raw",
+                HttpMethod.POST,
+                new HttpEntity<>(body, httpHeaders),
+                String.class)
+            .getBody();
+
+    var request = "{\"headers\":%s, \"body\":%s}";
+    try {
+      var headers = objectMapper.writeValueAsString(httpHeaders.entrySet());
+      request = String.format(request, headers, body);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    paymentDisbursementRepository.save(
+        new PaymentDisbursement("paymenthub", requestID, request, response));
+  }
+
+  @Override
+  @Transactional
+  public void updatePaymentOrderStatus(String callbackBody) throws JsonProcessingException {
+    JsonNode node = objectMapper.readTree(callbackBody);
+    var paymentDisbursement =
+        paymentDisbursementRepository.findByRequestUUID(node.get("requestId").asText());
+    paymentDisbursementRepository.save(paymentDisbursement.setCallback(callbackBody));
+  }
+
+  private String constructOrderPaymentRequestBody(List<Beneficiary> beneficiaryList) {
+    var rb =
+        beneficiaryList.stream()
+            .map(
+                beneficiary -> {
+                  return new PaymentHubOrderPaymentDTO(
+                      UUID.randomUUID().toString(),
+                      List.of(
+                          new PaymentHubOrderPaymentPartyDTO(
+                              beneficiary.getPerson().getFinancialModality().getKey(),
+                              beneficiary.getFunctionalId())),
+                      beneficiary.getPerson().getFinancialModality().paymentMode(),
+                      beneficiary.getEnrolledPackage().getAmount(),
+                      beneficiary.getEnrolledPackage().getCurrency(),
+                      "Payment for " + beneficiary.getEnrolledPackage().getName() + " package");
+                })
+            .toList();
+
+    try {
+      return objectMapper.writeValueAsString(rb);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private String signOrderPaymentRequest(String requestID, String body) {
+    var sha3 =
+        SHAUtils.generateSHA(
+            String.format("%s:%s:%s", requestID, paymentHubProperties.tenant(), body), "SHA3-256");
+    try {
+      return RSAUtil.encrypt(
+          sha3, RSAUtil.getPrivateKey(paymentHubProperties.jwsTenantPrivateKey()));
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
Just a note, usually one would use java.security.Signature and not the crypto primitives directly. Something like
```
var s = Signature.getInstance("SHA3-256withRSA");
s.initSign(<privatekey>)
s.update(...), s.sign(), ...
```